File: frontend/src/api/activity.ts
import { api } from './client';

export interface ActivityEvent {
  id: string;
  event_type: string;
  source: string;
  description: string;
  related_entity_type: string | null;
  related_entity_id: string | null;
  severity: string;
  created_at: string;
}

export function fetchActivity(limit = 20) {
  return api.get<ActivityEvent[]>(`/activity?limit=${limit}`);
}

File: backend/app/utils/activity.py
from sqlalchemy.ext.asyncio import AsyncSession

from app.database.models import ActivityEvent


async def record_event(
    db: AsyncSession,
    org_id: str,
    event_type: str,
    source: str,
    description: str,
    severity: str = "info",
    related_entity_type: str | None = None,
    related_entity_id: str | None = None,
):
    event = ActivityEvent(
        organization_id=org_id,
        event_type=event_type,
        source=source,
        description=description,
        severity=severity,
        related_entity_type=related_entity_type,
        related_entity_id=related_entity_id,
    )
    db.add(event)
